src/three/loaders/SUBTREELoader.js

```diff
@@ -0,0 +1,694 @@
+	 * @param {number} lengthBits The length of the availability bitstream in bits
+	 * @returns {ParsedBitstream}
+	 * @private
+	 */
+	parseAvailabilityBitstream(
+		availabilityJson,
+		bufferViewsU8,
+		lengthBits,
+	) {
+
+		if (!isNaN(availabilityJson.constant)) {
+			return {
+				constant: Boolean(availabilityJson.constant),
+				lengthBits: lengthBits,
+			};
+		}
+
+
+
+		let bufferView;
+
+		// Check for bitstream first, which is part of the current schema.
+		// bufferView is the name of the bitstream from an older schema.
+
+		if (!isNaN(availabilityJson.bitstream)) {
+			bufferView = bufferViewsU8[availabilityJson.bitstream];
+		} else if (!isNaN(availabilityJson.bufferView)) {
+			bufferView = bufferViewsU8[availabilityJson.bufferView];
+		}
+
+		return {
+			bitstream: bufferView,
+			lengthBits: lengthBits
+		};
+
+	}
+
+
+
+	/**
+	 * Expand a single subtree tile. This transcodes the subtree into
+	 * a tree of {@link SubtreeTile}. The root of this tree is stored in
+	 * the placeholder tile's children array. This method also creates
+	 * tiles for the child subtrees to be lazily expanded as needed.
+	 *
+	 * @param {Object | SubtreeTile} subtreeRoot The first node of the subtree
+	 * @param {Subtree} subtree The parsed subtree
+	 * @private
+	 */
+	expandSubtree(subtreeRoot, subtree) {
+
+		//	If the subtree root tile has content, then create a placeholder child with cloned parameters
+		//	Todo Multiple contents not handled, keep the first found content
+		for(let i = 0; subtree && i < subtree._contentAvailabilityBitstreams.length; i++){
+			if (subtree && this.getBit(subtree._contentAvailabilityBitstreams[i], 0)) {
+
+				// Create a child holding the content uri, this child is similar to its parent and doesn't have any children
+				let contentTile = {
+					content: {uri: parseImplicitURI(subtreeRoot, this.rootTile.__basePath, this.rootTile.__contentUri)},
+					boundingVolume: subtreeRoot.boundingVolume,
+					geometricError: subtreeRoot.geometricError
+				}
+				TilesRenderer.preprocessNode(contentTile, this.rootTile.__basePath, subtreeRoot?.parent);
+				subtreeRoot.children.push(contentTile);
+				break;
+			}
+		}
+
+		// Creating each leaf inside the current subtree
+		const bottomRow = this.transcodeSubtreeTiles(
+			subtreeRoot,
+			subtree
+		);
+
+		// For each child subtree, create a tile containing the uri of the next subtree to fetch
+		const childSubtrees = this.listChildSubtrees(subtree, bottomRow);
+		for (let i = 0; i < childSubtrees.length; i++) {
+			const subtreeLocator = childSubtrees[i];
+			const leafTile = subtreeLocator.tile;
+			let subtreeTile = this.deriveChildTile(
+				null,
+				leafTile,
+				null,
+				subtreeLocator.childMortonIndex
+			)
+			//	Assign subtree uri as content
+			subtreeTile.content = {uri: parseImplicitURI(subtreeTile, this.rootTile.__basePath, this.rootTile.__subtreeUri)};
+			TilesRenderer.preprocessNode(subtreeTile, this.rootTile.__basePath, leafTile);
```

Comment:
There's no need to call preprocessNode here - the children are preprocessed during traversal to avoid the expensive up front cost:

https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/b3206619ea2a9b20f8aa40c6db8842e69ef89349/src/base/TilesRendererBase.js#L403-L419

Reply:
The issue I've encountered is that we need to have some attributes defined when `preprocessNode` is called.
It looks like `children` needed to be defined, I can easily correct this by adding a children attribute in `SubtreeTile`.
Also, I'm using the `__depth` attribute, so I had to defined it in  `SubtreeTile`. By doing this, it will skip the call you're pointing at, because we will always fall in the condition and break.
I don't really know how to handle this gracefully... In the incoming commit, I introduce a "__level" attribute which is a duplicate from "__depth" :/ and patch this issue

Reply:
> It looks like children needed to be defined, I can easily correct this by adding a children attribute in SubtreeTile.

Is there a specific error you're seeing? As far as I can tell the `preprocessNode` function is resilient [to children not being set](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/4870d4f7e6195748638047422ebe0bf75f22da53/src/base/TilesRendererBase.js#L320) - as it needs to be since children isn't required by the [Tile schema](https://github.com/CesiumGS/3d-tiles/blob/main/specification/schema/tile.schema.json).

> I don't really know how to handle this gracefully... In the incoming commit, I introduce a "__level" attribute which is a duplicate from "__depth" :/ and patch this issue

I see this value is being used to calculate the implicit size of the child bounds. Lets use `__level` as an internal field to the SubTreeLoader for now and remove references to `__depth` and we can reevaluate later. Ideally the logic in SubTreeLoader should be self contained and not rely on side effects or methods from the TilesRenderer.

It's also the case that implicit tiling can be added to _any_ child in the tree in which case `__depth` will not start at 0 at that tile so it's probably not the right value to be relying on, anyway. From the spec:

> An implicitTiling object may be added to any tile in the tileset JSON.

Reply:
> Is there a specific error you're seeing? As far as I can tell the preprocessNode function is resilient [to children not being set](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/4870d4f7e6195748638047422ebe0bf75f22da53/src/base/TilesRendererBase.js#L320) - as it needs to be since children isn't required by the [Tile schema](https://github.com/CesiumGS/3d-tiles/blob/main/specification/schema/tile.schema.json).

I need my tiles to have children (for example [here](https://github.com/NASA-AMMOS/3DTilesRendererJS/pull/650/files#diff-24e9f963ffe77ddc637c982e940d4d8f021e51bf86ab27d170afac4a9599e38aR479) ), they are not preprocessed yet

> I see this value is being used to calculate the implicit size of the child bounds. Lets use __level as an internal field to the SubTreeLoader for now and remove references to __depth and we can reevaluate later. Ideally the logic in SubTreeLoader should be self contained and not rely on side effects or methods from the TilesRenderer.

Yes exactly, the geometricError and the BoundingBox depends on the depth/level and parent value

> It's also the case that implicit tiling can be added to any child in the tree in which case __depth will not start at 0 at that tile so it's probably not the right value to be relying on, anyway. From the spec:

This case hasn't been implemented. I'm only considering the data of the `root` object in the tileset, not the possible children.
My implementation is like a first draft to have a structure of implicit tiling, i've listed the current limitation. 
My idea is that, the support of implicit tiling is a big chunk, that should be sliced in smaller chunks. This MR could be the base to build a complete support of implicit tiling but it is not a full support :building_construction: